File: apps/web/src/infrastructure/repositories/category.repository.impl.ts
import { apiFetch } from "@/lib/api";
import { Category } from "../../domain/models/category";
import { CategoryRepository } from "../../domain/repositories/category.repository";

export class CategoryRepositoryImpl implements CategoryRepository {
    async list(): Promise<Category[]> {
        return apiFetch("/categories");
    }
}

File: apps/api/app/infrastructure/database/repositories/category_repository_impl.py
from typing import List
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select

from app.domain.entities.professional import Category as DomainCategory
from app.domain.repositories.category_repository import CategoryRepository
from app.models.category import Category as CategoryModel

class CategoryRepositoryImpl(CategoryRepository):
    def __init__(self, db: AsyncSession):
        self.db = db

    async def list_active(self) -> List[DomainCategory]:
        query = (
            select(CategoryModel)
            .where(CategoryModel.is_active == True)
            .order_by(CategoryModel.sort_order, CategoryModel.name)
        )
        result = await self.db.execute(query)
        models = result.scalars().all()
        return [
            DomainCategory(id=m.id, name=m.name, color=m.color)
            for m in models
        ]
